colorls.rb

```diff
@@ -1,23 +1,68 @@
+#!/usr/bin/env ruby
+
 require 'colorize'
 require 'yaml'
 require 'facets'
 require 'terminfo'
 
 # Source for icons unicode: http://nerdfonts.com/
-class ColorLS
-  def initialize(input, report)
+class ColorLS # rubocop:disable ClassLength
+  def initialize(input, report, sort, one_per_line)
     @input        = input || Dir.pwd
-    @contents     = Dir.entries(@input) - ['.', '..']
     @count        = { folders: 0, recognized_files: 0, unrecognized_files: 0 }
     @report       = report
+    @sort         = sort
+    @one_per_line = one_per_line
     @screen_width = TermInfo.screen_size.last
-    @max_widths   = @contents.map(&:length)
+
+    init_contents
+
+    @max_widths = @contents.map(&:length)
 
     init_icons
   end
 
+  def init_contents
+    if Dir.exist?(@input)
+      @contents = Dir.entries(@input) - ['.', '..']
+    elsif File.exist?(@input)
+      @contents = [@input]
+    else
+      raise ArgumentError, "Specified path doesn't exist: " + @input
+    end
```

Comment:
If you'd also like to support `lc *.rb`, we could definitely have these functionalities in a separate PR using regexp to show only matching files. Can you revert support for files input in this PR? (ie, please revert init_contents method, as specified above) 😄